service-base/src/test/java/org/eclipse/hono/service/credentials/AbstractCredentialsServiceTest.java

```diff
@@ -702,16 +700,15 @@ public void testDisableCredentials(final VertxTestContext ctx) {
 
         // create device
 
-        final Promise<?> phase1 = Promise.promise();
-        getDeviceManagementService().createDevice(
-                tenantId, Optional.of(deviceId), new Device(), NoopSpan.INSTANCE,
-                ctx.succeeding(s -> phase1.complete()));
+        final Future<OperationResult<Id>> phase1Future = getDeviceManagementService().createDevice(
+                tenantId, Optional.of(deviceId), new Device(), NoopSpan.INSTANCE)
+                .setHandler(ctx.succeeding(s -> Future.succeededFuture(s)));
```

Comment:
It looks to me as if the
`.setHandler(ctx.succeeding(s -> Future.succeededFuture(s)))`
part can be removed. 

The future returned by `createDevice` gets a handler here and is assigned to `phase1Future`.
Some lines below the `phase1Future` gets a handler again, overwriting the previous one. Therefore the above line looks superfluous.

Same applies to 7 other occurrences of a similar block in this class.

Reply:
@calohmn Done